config/config.go

```diff
@@ -68,18 +97,80 @@ func LoadConfig(filename string) (*Config, error) {
 	if err := yaml.UnmarshalStrict(confContent, &cfg); err != nil {
 		return nil, xerrors.Errorf("unable to parse config: %w", err)
 	}
+	if cfg.SchemaFilename != nil && cfg.Endpoint != nil {
+		return nil, errors.New("'schema' and 'endpoint' both specified. Use schema to load from a local file, use endpoint to load from a remote server (using introspection)")
+	} else if cfg.SchemaFilename == nil && cfg.Endpoint == nil {
+		return nil, errors.New("neither 'schema' nor 'endpoint' specified. Use schema to load from a local file, use endpoint to load from a remote server (using introspection)")
+	}
+
+	preGlobbing := cfg.SchemaFilename
+	cfg.SchemaFilename = []string{}
```

Comment:
zero value of slice is nil. but []string{} is pointer of length 0 array. So `c.SchemaFilename != nil` must be true. 

https://play.golang.org/p/-qRJrcHbvaW

Reply:
https://github.com/Yamashou/gqlgenc/pull/20#issuecomment-727725344

Reply:
I fix it with just idea.

https://github.com/Yamashou/gqlgenc/commit/f06757c9b749e488d955509e101d84cd76fd62af

Reply:
I apologise for the inconvenient.
But there's something strange, I've been using this fork with both local and remote schema for a while and this problem didn't occur to me.
Moreover you are right: obviously `[]string{} != nil` but I'm wondering where the slice is initialized, because is shouldn't be if `schema` does not exists in the config.
I'll deep dive into this and I'll follow up with you asap.

Reply:
Hi @Yamashou, I'm back to you with the answear.
The problem is not what you thought: here
https://github.com/FedeBev/gqlgenc/blob/f64cacbcc6fa4b048fbdbfc8503b44c68aa066b2/config/config.go#L100
`cfg.SchemaFilename` is nill as expected, the mistake is here
https://github.com/FedeBev/gqlgenc/blob/f64cacbcc6fa4b048fbdbfc8503b44c68aa066b2/config/config.go#L187
because `c.SchemaFilename` in no loger nil as it was initialized here
https://github.com/FedeBev/gqlgenc/blob/f64cacbcc6fa4b048fbdbfc8503b44c68aa066b2/config/config.go#L107

Long story short:
```go
	if c.SchemaFilename != nil {
		s, err := c.loadLocalSchema()
		if err != nil {
			return xerrors.Errorf("load local schema failed: %w", err)
		}
		schema = s
	} ellse {
```
should be
```go
	if len(c.SchemaFilename) != 0 {
		s, err := c.loadLocalSchema()
		if err != nil {
			return xerrors.Errorf("load local schema failed: %w", err)
		}
		schema = s
	} else {
```

That's it

In my opinion f06757c is not the best fix because it could be confusing and should be reverted in favour of the above fix.
Sorry again for my mistake and thanks for your work

Reply:
Even better, we can keep the second part of your fix https://github.com/Yamashou/gqlgenc/blob/f06757c9b749e488d955509e101d84cd76fd62af/config/config.go#L109-L141
but I'd revert the first part https://github.com/Yamashou/gqlgenc/blob/f06757c9b749e488d955509e101d84cd76fd62af/config/config.go#L101-L107
keeping `cfg.SchemaFilename == nil`.
Doing that way, `LoadSchema` does not require any fix

Reply:
Thanks your report! 

>keeping cfg.SchemaFilename == nil.
Doing that way, LoadSchema does not require any fix

I understand. it's great !
Checking for length is like safety net for bug. if we understand bug, should revert it.
Do you fix it ?

Reply:
> Do you fix it ?
I've just pushed the fix into develop branch of my fork, since this one is closed do you wish I open an other PR?

Reply:
Yes! I want it!